hedera-node/src/main/java/com/hedera/services/sigs/order/HederaSigningOrder.java

```diff
@@ -550,11 +551,18 @@ private boolean hasNondeprecatedAdminKey(ContractUpdateTransactionBody op) {
 
 		KeyOrderingFailure failure;
 		for (TokenTransferList xfers : op.getTokenTransfersList()) {
+			// fungible tokens
 			for (AccountAmount adjust : xfers.getTransfersList()) {
 				if ((failure = includeIfPresentAndNecessary(adjust, required)) != NONE) {
 					return accountFailure(adjust.getAccountID(), txnId, failure, factory);
 				}
 			}
+			// non fungible tokens
+			for (NftTransfer adjust : xfers.getNftTransfersList()) {
+				if ((failure = includeIfPresentAndNecessary(adjust, required)) != NONE) {
```

Comment:
We should change this logic here because if there is a failure we do not know what account caused the failure -> whether it is the sender or receiver account. For fungibles/hbars there is no problem since account amount has `account and amount` properties, but in our case that is not the case. We should refactor the logic such that we have independent calls to `includeIfPresentAndNecessary` for sender and receiver. If one fails we will know which account to pass to `accountFailure`

Reply:
Confirmed and fixed!